src/core/sys/linux/sys/socket.d

```diff
@@ -13,7 +13,146 @@ extern(C):
 @nogc:
 nothrow:
 
-enum {
-    AF_RXRPC    = 33,
-    PF_RXRPC    = AF_RXRPC,
+enum
+{
+    // Protocol families.
+    PF_UNSPEC     = 0,
+    PF_LOCAL      = 1,
+    PF_UNIX       = PF_LOCAL,
+    PF_FILE       = PF_LOCAL,
+    PF_INET       = 2,
+    PF_AX25       = 3,
+    PF_NETROM     = 6,
+    PF_BRIDGE     = 7,
+    PF_ATMPVC     = 8,
+    PF_X25        = 9,
+    PF_INET6      = 10,
+    PF_ROSE       = 11,
+    PF_DECnet     = 12,
+    PF_NETBEUI    = 13,
+    PF_SECURITY   = 14,
+    PF_KEY        = 15,
+    PF_NETLINK    = 16,
+    PF_ROUTE      = PF_NETLINK,
+    PF_PACKET     = 17,
+    PF_ASH        = 18,
+    PF_ECONET     = 19,
+    PF_ATMSVC     = 20,
+    PF_RDS        = 21,
+    PF_SNA        = 22,
+    PF_IRDA       = 23,
+    PF_PPPOX      = 24,
+    PF_WANPIPE    = 25,
+    PF_LLC        = 26,
+    PF_IB         = 27,
+    PF_MPLS       = 28,
+    PF_CAN        = 29,
+    PF_TIPC       = 30,
+    PF_BLUETOOTH  = 31,
+    PF_IUCV       = 32,
+    PF_RXRPC      = 33,
+    PF_ISDN       = 34,
+    PF_PHONET     = 35,
+    PF_IEEE802154 = 36,
+    PF_CAIF       = 37,
+    PF_ALG        = 38,
+    PF_NFC        = 39,
+    PF_VSOCK      = 40,
+    PF_KCM        = 41,
+    PF_QIPCRTR    = 42,
+    PF_SMC        = 43,
+    PF_MAX        = 44,
```

Comment:
At https://github.com/torvalds/linux/blob/master/include/linux/socket.h `PF_MAX` is 45. Where is your list coming from?

Reply:
Linux 4.15, which is not the latest release, but still fairly new.

Reply:
https://github.com/torvalds/linux/blob/v4.17/include/linux/socket.h#L211

So the new macro was introduced in 4.18, which is the current release.

Reply:
> Neither 4.15 nor 4.18 socket.h seem to have AF_FILE/PF_FILE. Where is it coming from? AF_IPX/PF_IPX and AF_APPLETALK/PF_APPLETALK are present in 4.15 & 4.18 socket.h but are not in this PR. Why?

You're looking at the wrong header. It's in glibc / musl / insert your C library here headers.

Reply:
Is it a problem that `PF_NETLINK` and `PF_ROUTE` are defined as 16 here and in linux/musl/uClibc's headers but glibc's socket.h has no `PF_NETLINK` (instead having `#define	PF_APPLETALK	16	/* Don't use this.  */`) and defines `PF_ROUTE` as 17?

Reply:
> AF_IPX/PF_IPX and AF_APPLETALK/PF_APPLETALK are present in 4.15 & 4.18 socket.h but are not in this PR. Why?

Check the posix.sys.socket module. At least APPLETALK is present there, although it perhaps shouldn't be as it's nonstandard.

Reply:
> Is it a problem that PF_NETLINK and PF_ROUTE are defined as 16 here and in linux/musl/uClibc's headers but glibc's socket.h has no PF_NETLINK (instead having #define PF_APPLETALK 16 /* Don't use this. */) and defines PF_ROUTE as 17?

I don't think so.  Can check what value ROUTE would be on glibc at runtime. If its already in Linux specific headers, then it shouldn't be overridden by libc.